public/vendor/flot/jquery.flot.gauge.js

```diff
@@ -344,20 +343,48 @@
 
             // draw gauge
             var c1 = getColor(gaugeOptionsi, data);
-            var a2 = calculateAngle(gaugeOptionsi, layout, data);
+            var angles = calculateAnglesForGauge(gaugeOptionsi, layout, data);
+
             drawArcWithShadow(
                 cellLayout.cx, // center x
                 cellLayout.cy, // center y
                 layout.radius - 1,
                 layout.width - 2,
-                toRad(gaugeOptionsi.gauge.startAngle),
-                toRad(a2),
+                toRad(angles.a1),
+                toRad(angles.a2),
                 c1,           // line color
                 1,            // line width
                 c1,           // fill color
                 blur);
         }
 
+
+        /**
+         * if gauge min is negative and has the absolute value of min and max are the same
+         * then calculate a1 and a2 in a way, that zero is at 12 o'clock
+         * 
+         * @method calculateAnglesForGauge
+         * @param {Object} gaugeOptionsi the options of the gauge
+         * @returns {Object}
+         */
+        function calculateAnglesForGauge(gaugeOptionsi, layout, data) {
+            let angles = {};
+            var setNorth = (gaugeOptionsi.gauge.min + gaugeOptionsi.gauge.max) == 0.0
+            
+            angles.a1 = gaugeOptionsi.gauge.startAngle;
+            angles.a2 = calculateAngle(gaugeOptionsi, layout, data);
+
+            if(setNorth && angles.a2<=1.5) {
+                angles.a1 = angles.a2;
+                angles.a2 = 1.5;
+            }
+            if(setNorth && angles.a2>1.5) {
+                angles.a1 = 1.5;
+            }
+            
+            return angles;
+        }
```

Comment:
Hey @sfranzis thank you for contributing to Grafana! 🎉 

I have to agree with the sentence "limiting to inverse min/max scenario seems arbitrary", I think it would end up being almost a "hidden feature" that if you set min and max to the same absolute value you get this extra feature on the Gauges that have negative numbers.

I think ideally this should always happen if the minimum of the gauge is < 0, and then we should be able to see some kind of indication of where 0 is as well, maybe with a dotted line or with the actual number 0 next to that point.
So I feel like in order for us to merge this it would need a little bit more work. This is my suggestion for what this function should look like:
```
        /**
         * Calcualte the angles for the gauge, depending on if there are
         * negative numbers or not.
         * 
         * @method calculateAnglesForGauge
         * @param {Object} gaugeOptionsi the options of the gauge
         * @param  {Number} data the value of the gauge
         * @returns {Object}
         */
        function calculateAnglesForGauge(gaugeOptionsi, layout, data) {
            let angles = {};
            var hasNegative = gaugeOptionsi.gauge.min < 0.0
            var isNegative = data < 0.0
            
            if (hasNegative) {
                if (isNegative) {
                    angles.a1 = calculateAngle(gaugeOptionsi, layout, data);
                    angles.a2 = calculateAngle(gaugeOptionsi, layout, 0.0);
                } else {
                    angles.a1 = calculateAngle(gaugeOptionsi, layout, 0.0);
                    angles.a2 = calculateAngle(gaugeOptionsi, layout, data);
                }
            } else {
                angles.a1 = gaugeOptionsi.gauge.startAngle;
                angles.a2 = calculateAngle(gaugeOptionsi, layout, data);
            }
            
            return angles;
        }
```

What do you think? Then the only thing left would be to find a way to represent where the 0 is when the min is negative. Would you like to give that a go?

Reply:
Hey @JoaoSilvaGrafana thanks for your suggestion. I'm fine with it. 👍